packages/webpack-cli/lib/plugins/InteractiveModePlugin.js

```diff
@@ -0,0 +1,207 @@
+
+        // Register Custom Hook for printing after clrscr
+        if (!compiler.hooks.beforeInteractiveOutput) {
+            compiler.hooks = {
+                ...compiler.hooks,
+                beforeInteractiveOutput: new SyncHook(),
+            };
+        }
+
+        let beforeCompileCount = 0;
+        for (const childCompiler of compilers) {
+            // eslint-disable-next-line no-loop-func
+            childCompiler.hooks.beforeCompile.tap(this.name, () => {
+                if (beforeCompileCount === 0) {
+                    clrscr();
+                    compiler.hooks.beforeInteractiveOutput.call();
+                }
+
+                beforeCompileCount += 1;
+                if (beforeCompileCount === compilers.length) {
+                    beforeCompileCount = 0;
+                }
+            });
+        }
+
+        let afterDoneCount = 0;
+        for (const childCompiler of compilers) {
+            // eslint-disable-next-line no-loop-func
+            childCompiler.hooks.afterDone.tap(this.name, () => {
+                afterDoneCount += 1;
+                if (afterDoneCount === compilers.length) {
+                    afterDoneCount = 0;
+                    process.nextTick(() => {
+                        process.stdout.write(`${this.verbose ? "\n" : ""}\n\n`);
+                        spawnCommand("compilations completed", true, false, this.verbose);
+                    });
+                }
+            });
+        }
+    }
+
+    quitHandler(_compilers, compiler) {
+        compiler.close(() => {
+            spawnCommand("", true, true, this.verbose);
+            // Show cursor
+            process.stdout.write("\u001B[?25h");
+            process.exit(0);
+        });
+    }
+
+    startHandler(compilers) {
+        const allWatching = compilers.reduce((result, childCompiler) => {
+            return result && !childCompiler.watching.suspended;
+        }, true);
+
+        if (allWatching) {
+            spawnCommand("already watching", true, false, this.verbose);
+            return;
+        }
+
+        for (const childCompiler of compilers) {
+            if (childCompiler.watching && childCompiler.watching.suspended) {
+                childCompiler.watching.resume();
+                childCompiler.compile(() => {});
+            }
+        }
+    }
+
+    stopHandler(compilers) {
+        const allSuspended = compilers.reduce((result, childCompiler) => {
+            return result && childCompiler.watching.suspended;
+        }, true);
+
+        if (allSuspended) {
+            spawnCommand("already stoped", false, false, this.verbose);
+            return;
+        }
+
+        for (const childCompiler of compilers) {
+            if (!childCompiler.watching.suspended) {
+                childCompiler.watching.suspend();
+            }
+        }
+        spawnCommand("stoped watching", false, false, this.verbose);
+        return;
+    }
+}
+module.exports = InteractiveModePlugin;
```

Comment:
A lot of uncovered branches let's add more tests

Reply:
Will do.

Reply:
Updated tests, let's see coverage once Ci is green.

Reply:
Coverage<img width="1114" alt="Screen Shot 2021-06-25 at 11 25 08 AM" src="https://user-images.githubusercontent.com/15812317/123376433-17799c80-d5a8-11eb-8bb3-b36c6668964d.png">is now fine I guess.